query/src/provider.rs

```diff
@@ -3167,6 +3167,201 @@ mod test {
+                .with_tag_column("tag1")
+                .with_i64_field_column("field_int")
+                .with_partition_id(1) // signal the chunk in partition 1 while grouping overlaps
+                .with_may_contain_pk_duplicates(true) // signal having duplicates within this chunk
+                .with_sort_key(sort_key.clone()), // signal the chunk is sorted
+        ) as Arc<dyn QueryChunk>;
+
+        // ------------
+        // Partition 2: Same 4 kinds of chunks on 6 chunks
+        let chunk2_1 = Arc::new(
+            TestChunk::new("t")
+                .with_id(1)
+                .with_time_column()
+                .with_timestamp_min_max(5, 7000)
+                .with_tag_column("tag1")
+                .with_i64_field_column("field_int")
+                .with_partition_id(2) // signal the chunk in partition 2
+                .with_sort_key(sort_key.clone()), // signal the chunk is sorted
+        ) as Arc<dyn QueryChunk>;
+
+        // chunk2_2 overlaps with chunk 2_1
+        let chunk2_2 = Arc::new(
+            TestChunk::new("t")
+                .with_id(2)
+                .with_time_column()
+                .with_timestamp_min_max(5, 7000)
+                .with_tag_column("tag1")
+                .with_i64_field_column("field_int")
+                .with_partition_id(2) // signal the chunk in partition 2
+                .with_sort_key(sort_key.clone()), // signal the chunk is sorted
+        ) as Arc<dyn QueryChunk>;
+
+        // chunk2_2 no overlap, no duplicates within
+        let chunk2_3 = Arc::new(
+            TestChunk::new("t")
+                .with_id(3)
+                .with_time_column()
+                .with_timestamp_min_max(8000, 20000)
+                .with_tag_column("tag1")
+                .with_i64_field_column("field_int")
+                .with_partition_id(2) // signal the chunk in partition 2
+                .with_sort_key(sort_key.clone()), // signal the chunk is sorted
+        ) as Arc<dyn QueryChunk>;
+
+        // chunk2_4 no overlap, no duplicates within
+        let chunk2_4 = Arc::new(
+            TestChunk::new("t")
+                .with_id(3)
+                .with_time_column()
+                .with_timestamp_min_max(21000, 27000)
+                .with_tag_column("tag1")
+                .with_i64_field_column("field_int")
+                .with_partition_id(2) // signal the chunk in partition 2
+                .with_sort_key(sort_key.clone()), // signal the chunk is sorted
+        ) as Arc<dyn QueryChunk>;
+
+        // chunk2_5 no overlap, duplicates within
+        let chunk2_5 = Arc::new(
+            TestChunk::new("t")
+                .with_id(4)
+                .with_time_column()
+                .with_timestamp_min_max(28000, 30000)
+                .with_tag_column("tag1")
+                .with_i64_field_column("field_int")
+                .with_partition_id(2) // signal the chunk in partition 1 while grouping overlaps
```

Comment:
Copypasta, I think this should be:

```suggestion
                .with_partition_id(2) // signal the chunk in partition 2 while grouping overlaps
```